src/plugins/discover/public/application/view_components/canvas/index.tsx

```diff
@@ -164,19 +163,20 @@ export default function DiscoverCanvas({ setHeaderActionMenu, history, optionalR
           {fetchState.status === ResultStatus.UNINITIALIZED && (
             <DiscoverUninitialized onRefresh={() => refetch$.next()} />
           )}
-          {fetchState.status === ResultStatus.LOADING && <LoadingSpinner />}
-          {fetchState.status === ResultStatus.READY && isEnhancementsEnabled && (
-            <>
-              <MemoizedDiscoverChartContainer {...fetchState} />
-              <MemoizedDiscoverTable rows={rows} scrollToTop={scrollToTop} />
-            </>
-          )}
-          {fetchState.status === ResultStatus.READY && !isEnhancementsEnabled && (
-            <EuiPanel hasShadow={false} paddingSize="none" className="dscCanvas_results">
-              <MemoizedDiscoverChartContainer {...fetchState} />
-              <MemoizedDiscoverTable rows={rows} scrollToTop={scrollToTop} />
-            </EuiPanel>
-          )}
+          {fetchState.status === ResultStatus.LOADING && !rows?.length && <LoadingSpinner />}
+          {(fetchState.status === ResultStatus.READY ||
+            (fetchState.status === ResultStatus.LOADING && !!rows?.length)) &&
```

Comment:
why do we have two different result for loading status now?

Reply:
we want to show results if query finished and results are ready, or if there are previous results while current query is loading